backend/src/main/java/com/bakdata/conquery/apiv1/QueryProcessor.java

```diff
@@ -264,9 +277,15 @@ private void translateToOtherDatasets(Dataset dataset, QueryDescription query, U
 		}
 	}
 
-	public void cancel(User user, Dataset dataset, ManagedExecution<?> query, UriBuilder urlb) {
-		// TODO implement query cancel functionality
-		return;
+	public void cancel(User user, Dataset dataset, ManagedExecution<?> query) {
+
+		log.debug("{} cancelled Query[{}]", user, query.getId());
+
+		final Namespace namespace = getDatasetRegistry().get(dataset.getId());
+
+		namespace.sendToAll(new CancelQuery(query.getId()));
+
+		query.setState(ExecutionState.CANCELED);
```

Comment:
Ich würde in Hinblick auf die Zusammenlegung von Query und StoredQueries, einfach die Execution auf Manager-Seite löschen. Im ExecutionManager#addQueryResult würde ich die empfangenen Result dann einfach verwerfen, wenn ich die Execution nicht finde.

Reply:
So würden bei einer gecancelten Execution dennoch alle Ergebnisse beim Manager eintrudeln und der State auf FINISHED springen.

Reply:
Ich mag den ExecutionState.CANCELED nicht, weil er aus meiner Sicht keinen Mehrwert bringt.

Reply:
Hier kann man auch FINISHED Executions noch auf CANCELED setzen 🤔

Reply:
Also löschen würde ich sie auf keinen Fall, durch reexecute und reuse gibt es ja die möglichkeit alte anfragen wieder auszuführen und ggf abzubrechen ohne sie komplett entfernen zu müssen.

Theoretisch kann ich den State auch auf NEW setzen, weil es effektiv nach außen genau so aussehen soll.

Reply:
Aber warum sollte man eine abgebrochene Query nochmals ausführen?

Reply:
Hatten wir besprochen, es kann vorkommen, dass es eine Reused Query ist und dann würde ich eine alte Anfrage löschen die ich bewusst gespeichert habe.